docs/docs/examples/vault.md

```diff
@@ -0,0 +1,264 @@
+
+:::
+<!-- TODO(jmxnzo): revise this paragraph as soon as we support filesystem volumes everywhere -->
+The Vault deployment is defined as a StatefulSet using the OpenBao Vault image, with a mounted block device for persistent storage.
+A Contrast Initializer, running as an init container, uses the workload secret located at `/contrast/secrets/workload-secret-seed` to
+generate an encryption key and initialize the block device as a LUKS-encrypted partition.
+Before the Vault container starts, the Initializer opens the LUKS device using the generated key.
+This unlocked device is then mounted by the Vault container and used as the backend storage volume.
+For the Vault application, this process is entirely transparent, and the device behaves like a standard volume mount.
+It’s important to clarify that the LUKS encryption of the block device is primarily a convenience feature, enabling persistent storage
+at the filesystem level on confidential virtual machines.
+The primary and security-relevant encryption mechanism remains Vault’s own sealing process, which provides cryptographic protection of
+secrets—even in the event that the underlying storage is compromised.
+
+Because the `workload-secret-seed` is derived from the associated `workloadSecretID`,
+any change to the `workloadSecretID` after the block device has been initialized will result in deriving an invalid encryption key,
+making the mounted block device undecryptable.
+
+:::note[Inter-deployment communication]
+
+The Contrast Coordinator issues mesh certificates after successfully validating workloads.
+These certificates can be used for secure inter-deployment communication.
+The Initializer sends an attestation report to the Coordinator,
+retrieves a service mesh certificate bound to it's provided public key, containing the certificate chain, as well as the current mesh CA cert.
+The Initializer then writes them to a `volumeMount`, allowing to build up the secure mTLS connections based on the service mesh.
+The received service mesh certificate also holds the certificate extension of the `workloadSecretID`,
+which is used to allow the authorization to a certain encryption key on the transit engine API.
+
+:::
+
+
+The Vault's TCP listener is configured to accept connections only from mesh certificates issued under the current state of the service mesh CA,
+effectively restricting communication to attested Contrast deployments.
+The Coordinator’s transit secrets engine API authorizes requests based on the `workloadSecretID`,
+which is embedded in a certificate extension and must match the target endpoint.
+As previously noted, updating the `workloadSecretID` after initializing the LUKS device will make it inaccessible,
+due to a mismatch in the derived encryption key.
+Therefore, it's critical to ensure that the `workloadSecretID` is correctly aligned with the intended endpoint
+specified in Vault’s sealing configuration before the first `contrast set` is executed.
```

Comment:
I feel that this paragraph only repeats content from prior sections. It should either concisely introduce a new concept, or be removed.